main.py

```diff
@@ -0,0 +1,291 @@
+async def handle_mentorship_request_claim_click(
+    body: dict[str, Any], context: AsyncBoltContext
+) -> None:
+    logger.info("STAGE: Processing a mentorship request claim...")
+    await handle_mentorship_request_claim(SlackActionRequestBody(**body), context)
+
+
+@app.action("reset_mentorship_request_claim")
+async def handle_mentorship_request_claim_reset_click(
+    body: dict[str, Any], context: AsyncBoltContext
+) -> None:
+    logger.info("STAGE: Processing a mentorship request claim reset...")
+    await handle_mentorship_request_claim_reset(SlackActionRequestBody(**body), context)
+
+
+@app.command("/new_join")
+@app.event("member_joined_channel")
+async def handle_new_member_join_event(
+    body: dict[str, Any], context: AsyncBoltContext
+) -> None:
+    logger.info("STAGE: Processing new member joining...")
+    if body["command"]:
+        await handle_new_member_join(SlackCommandRequestBody(**body), context)
+    else:
+        await handle_new_member_join(MemberJoinedChannelEvent(**body), context)
+
+
+@app.action("greet_new_user_claim")
+async def handle_greeting_new_user_claim_action(
+    context: AsyncBoltContext,
+    body: dict[str, Any],
+) -> None:
+    logger.info("STAGE: Processing new claim on new user for greetings...")
+    await handle_greeting_new_user_claim(SlackActionRequestBody(**body), context)
+
+
+@app.action("reset_greet_new_user_claim")
+async def handle_resetting_greeting_new_user_claim_action(
+    context: AsyncBoltContext, body: dict[str, Any]
+) -> None:
+    logger.info("STAGE: Resetting claim on new user greeting...")
+    await handle_resetting_greeting_new_user_claim(
+        SlackActionRequestBody(**body), context
+    )
+
+
+@app.command("/report")
+async def handle_report_command(
+    body: dict[str, Any], context: AsyncBoltContext
+) -> None:
+    logger.info("STAGE: Processing report command...")
+    await handle_report(body, context)
+
+
+@app.view("report_form_submit")
+async def handle_report_view_submit(
+    body: dict[str, Any], context: AsyncBoltContext
+) -> None:
+    logger.info("STAGE: Processing report view submission...")
+    await handle_report_submit(body, context)
+
+
+@app.action("report_claim")
+async def handle_report_claim_action(
+    body: dict[str, Any], context: AsyncBoltContext
+) -> None:
+    logger.info("STAGE: Processing report claim....")
+    await handle_report_claim(
+        SlackResponseBody(**body, originating_user=SlackUserInfo(**body["user"])),
+        context,
+    )
+
+
+@app.action("reset_report_claim")
+async def handle_reset_report_claim_action(
+    body: dict[str, Any], context: AsyncBoltContext
+) -> None:
+    logger.info("STAGE: Processing reset of report claim...")
+    await handle_reset_report_claim(
+        SlackResponseBody(**body, originating_user=SlackUserInfo(**body["user"])),
+        context,
+    )
+
+
+@app.command("/join-pride")
+@app.command("/join-blacks-in-tech")
+async def handle_channel_join_request_command(
```

Comment:
I love the reusability here ❤️  I could see us adding similar functionality to other private channels where relevant. This is great